plugins/base/src/main/kotlin/translators/psi/DefaultPsiToDocumentableTranslator.kt

```diff
@@ -564,8 +564,6 @@ class DefaultPsiToDocumentableTranslator(
 
                 is PsiPrimitiveType -> if (type.name == "void") Void
                 else type.cacheBoundIfHasNoAnnotation { annotations -> PrimitiveJavaType(type.name, annotations.annotations()) }
-                is PsiImmediateClassType ->
```

Comment:
Not sure why it was done this way. Probably an exception was thrown and mapping `PsiImmediateClassType -> JavaObject` was added without researching.

`PsiImmediateClassType` and `PsiClassReferenceType` are both extending `PsiClassType` (`resolve()` is also part of the latter), so hopefully it's all good